com.sap.cloud.lm.sl.persistence/src/main/java/com/sap/cloud/lm/sl/persistence/services/ProcessLogsService.java

```diff
@@ -59,4 +69,36 @@ public String findFileId(String space, String processId, String fileName) throws
         return null;
     }
 
+    public int deleteAllByProcessIds(final List<String> processIds) throws SLException {
+        SqlExecutor<Integer> executor = new FileServiceSqlExecutor<Integer>();
+        try {
+            return executor.execute(new StatementExecutor<Integer>() {
+                @Override
+                public Integer execute(Connection connection) throws SQLException {
+                    int rowsRemoved = 0;
+                    PreparedStatement statement = null;
+                    try {
+                        connection.setAutoCommit(false);
+                        statement = connection.prepareStatement(getQuery(DELETE_CONTENT_BY_NAMESPACE, tableName));
+                        for (String processId : processIds) {
+                            statement.setString(1, processId);
+                            statement.addBatch();
+                        }
+                        int[] rowsRemovedArray = statement.executeBatch();
+                        rowsRemoved = Arrays.stream(rowsRemovedArray).sum();
```

Comment:
It would be easier to use just "rowsRemovedArray.length".

Reply:
Ok, I read that the rowsRemovedArray is 
_"an array of update counts containing one element for each command in the batch. The elements of the array are ordered according to the order in which commands were added to the batch."_

Then I assume your logic is correct 👍

Reply:
Ok, so now we leave it as it is.